added vectorized temperature parameter and assigned it to tank segmen…

JayHuLBL opened the pull request:
This closes #1246.

mwetter commented:
@JayHuLBL Please also make sure that there is a unit test that uses non-uniform initial tank temperature.

JayHuLBL commented:
@mwetter It will be ready for review after CI tests passed.
- [x] added tank validation model with non-uniform initial temperature
- [x] run unit test and added reference results
